### PR TITLE
Refine 'make cover' to cover other packages

### PR DESCRIPTION
**Pull request by aidan-kwon**
## Proposed changes

Current coverage test covers only the same package code on each test package. This PR introduces the following changes.
1. The change adds `-coverpkg` option which specifies package list covered by the test. 
2. ~`go test -cover` does not generate a single profile for test packages when it has `-coverpkg` option. To mitigate the issue, the `for loop` in the introduced function executes coverage test for each package and combines all test results in a single file, `coverage.out`.~

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [ ] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

Some packages might be excluded in the coverage test. 
Reviewers, add more package names to be excluded in the test.
For now, only `tests` is excluded.

### CODE REVIEW

File: build/ci.go [hunk 1]
```diff
@@ -272,6 +274,56 @@ func doTest(cmdline []string) {
...
+	packages := []string{"./..."}
+	if len(flag.CommandLine.Args()) > 0 {
+		packages = flag.CommandLine.Args()
+	}
+	packages = build.ExpandPackagesNoVendor(packages)
+	if *excludes != "" {
+		packages = build.ExcludePackages(packages, strings.Split(*excludes, ","))
+	}
+
+	coverPackages := []string{"./..."}
```
Comment (aidan-kwon) [inline build/ci.go]: `packages` in line 286 means the packages which have test codes to be executed.
`coverPackages` means the packages which have source codes to be investigated.

File: build/ci.go [hunk 2]
```diff
@@ -272,6 +274,56 @@ func doTest(cmdline []string) {
...
+	coverExcludes := []string{
+		"github.com/klaytn/klaytn/tests",
```
Comment (jimni1222) [inline build/ci.go]: How about adding "github.com/klaytn/klaytn/tests/benchmarks"?

File: build/ci.go [hunk 3]
```diff
@@ -272,6 +274,56 @@ func doTest(cmdline []string) {
...
+			gotest.Args = append(gotest.Args, "-p", strconv.Itoa(*parallel))
+		}
+
+		tmpFile := "coverage.tmp." + strconv.Itoa(i)
+
+		gotest.Args = append(gotest.Args, "-cover", "-covermode=atomic", "-coverprofile="+tmpFile, testPkg)
+		gotest.Args = append(gotest.Args, "-coverpkg", coverPackagesString)
+		build.MustRun(gotest)
+
+		build.MustRunCommand("sh", "-c", "tail -n +2 "+tmpFile+" >> "+*outputFile)
```
Comment (jimni1222) [inline build/ci.go]: Can you explain what this line means?

File: Makefile [hunk 4]
```diff
@@ -83,7 +83,7 @@ test-others:
 	build/env.sh go run build/ci.go test -p 1 -exclude datasync,networks,tests
 
 cover:
-	build/env.sh go run build/ci.go test -coverage
+	build/env.sh go run build/ci.go cover -coverprofile=coverage.out
```
Comment (ehnuje) [inline Makefile]: As you created an independent function `cover` for coverage test, how about cleaning-up coverage-related codes from `doTest`?

File: build/ci.go (hunk 2 above)
Comment (KimKyungup) [reply build/ci.go]: `vender` is already excluded?

Comment (aidan-kwon) [reply build/ci.go]: @jimni1222 I replaced the exclude pattern to `/test`.
@KimKyungup `vendor` is excluded in `ExpandPackagesNoVendor`.

File: Makefile (hunk 4 above)
Comment (aidan-kwon) [reply Makefile]: I cleaned up. Thank you!

File: build/ci.go [hunk 5]
```diff
@@ -264,10 +265,54 @@ func doTest(cmdline []string) {
...
+	if *excludes != "" {
+		packages = build.ExcludePackages(packages, strings.Split(*excludes, ","))
+	}
+
+	coverPackages := []string{"./..."}
+	coverExcludes := []string{
+		"/tests",
+		"/metric",
+		"/build",
+		"/contracts",
```
Comment (jimni1222) [inline build/ci.go]: Will we exclude contracts?

Comment (aidan-kwon) [reply build/ci.go]: removed it

File: build/ci.go [hunk 6]
```diff
@@ -264,10 +265,53 @@ func doTest(cmdline []string) {
...
+	if *excludes != "" {
+		packages = build.ExcludePackages(packages, strings.Split(*excludes, ","))
+	}
+
+	coverPackages := []string{"./..."}
+	coverExcludes := []string{
+		"/tests",
+		"/metric",
+		"/build",
+		"/client",
```
Comment (jeonghonah) [inline build/ci.go]: How about adding `/contracts`? PR #113 will move a test file to node/sc directory.

Comment (aidan-kwon) [reply build/ci.go]: Thank you for the recommendation. We are considering the option also. The excluding list will be discussed in other PR later.

**Pull request merged**